baselines/diabetic_retinopathy_detection/jax_finetune_deterministic.py

```diff
@@ -14,147 +14,89 @@
-    'you should specify an output_dir name that includes the random seed to '
-    'avoid overwriting.')
-flags.DEFINE_string(
-    'distribution_shift', None,
-    'Specifies distribution shift to use, if any.'
-    'aptos: loads APTOS (India) OOD validation and test datasets. '
-    '  Kaggle/EyePACS in-domain datasets are unchanged.'
-    'severity: uses DiabeticRetinopathySeverityShift dataset, a subdivision '
-    '  of the Kaggle/EyePACS dataset to hold out clinical severity labels '
-    '  as OOD.')
-flags.mark_flag_as_required('distribution_shift')
-flags.DEFINE_string(
-    'pretrain_dataset', 'imagenet21k',
-    'Dataset for model pretraining. Specifies the config to use.')
-flags.DEFINE_string(
-    'resume_checkpoint_path', None,
-    'If provided, resume training and/or conduct evaluation using this '
-    'checkpoint. Will only be used if the output_dir does not already '
-    'contain a checkpointed model. See checkpoint_utils.py.')
-
-# Logging and hyperparameter tuning.
-flags.DEFINE_bool('use_wandb', False, 'Use wandb for logging.')
-flags.DEFINE_string('wandb_dir', 'wandb', 'Directory where wandb logs go.')
-flags.DEFINE_string('project', 'ub-debug', 'Wandb project name.')
-flags.DEFINE_string('exp_name', None, 'Give experiment a name.')
-flags.DEFINE_string('exp_group', None, 'Give experiment a group name.')
-
-# Learning rate / SGD flags.
-flags.DEFINE_float('grad_clip_norm', 20.0, 'Gradient clipping threshold.')
-flags.DEFINE_float('weight_decay', None, 'Gradient clipping threshold.')
-flags.DEFINE_float('lr_base', 0.1, 'Base learning rate.')
-flags.DEFINE_integer('lr_warmup_steps', 500, 'Number of LR warmup steps.')
-flags.DEFINE_string('lr_decay_type', 'cosine',
-                    'Type of LR decay / schedule. Options: cosine, linear.')
-
-# General model flags.
-flags.DEFINE_string(
-    'vit_model_size', None,
-    'Specifies size of ViT backbone model. Options: {"B/16", "L/32"}')
-flags.mark_flag_as_required('vit_model_size')
-flags.DEFINE_integer('total_steps', 10000, 'Total steps.')
-flags.DEFINE_integer('batch_size', 128, 'Batch size.')
-flags.DEFINE_integer('seed', 0, 'Random seed.')
-flags.DEFINE_string(
-    'class_reweight_mode', None,
-    'Dataset is imbalanced (19.6%, 18.8%, 19.2% positive examples in train, '
-    'val, test respectively). `None` (default) will not perform any loss '
-    'reweighting. `constant` will use the train proportions to reweight the '
-    'binary cross entropy loss. `minibatch` will use the proportions of each '
-    'minibatch to reweight the loss.')
-
-# Evaluation flags.
-flags.DEFINE_bool('only_eval', False,
-                  'Disables training, only evaluates the model.')
-flags.DEFINE_bool('use_validation', True,
-                  'Whether to use a validation split.')
-flags.DEFINE_bool('use_test', True, 'Whether to use a test split.')
+logging.info(tf.config.experimental.get_visible_devices())
 
+ml_collections.config_flags.DEFINE_config_file(
```

Comment:
Following above comment about `total_steps` and `warmup_steps`, let's keep the config locked.